pack/app_dir.go

```diff
@@ -0,0 +1,269 @@
+	versionFileLines = append(versionFileLines, appVersionLine)
+
+	// Tarantool version
+	if projectCtx.TarantoolIsEnterprise {
+		tarantoolVersionFilePath := filepath.Join(projectCtx.TarantoolDir, "VERSION")
+		tarantoolVersionFile, err := os.Open(tarantoolVersionFilePath)
+		defer tarantoolVersionFile.Close()
+
+		if err != nil {
+			log.Warnf("Can't open VERSION file from Tarantool SDK: %s. SDK information can't be "+
+				"shipped to the resulting package. ", err)
+		}
+
+		scanner := common.FileLinesScanner(tarantoolVersionFile)
+		for scanner.Scan() {
+			versionFileLines = append(versionFileLines, scanner.Text())
+		}
+	} else {
+		tarantoolVersionLine := fmt.Sprintf("TARANTOOL=%s", projectCtx.TarantoolVersion)
+		versionFileLines = append(versionFileLines, tarantoolVersionLine)
+	}
+
+	// rocks versions
+	rocksVersionsMap, err := getRocksVersions(appDirPath, projectCtx)
+	if err != nil {
+		log.Warnf("can't process rocks manifest file. Dependency information can't be "+
+			"shipped to the resulting package: %s", err)
+	} else {
+		for rockName, rockVersion := range rocksVersionsMap {
+			if rockName != projectCtx.Name {
+				rockLine := fmt.Sprintf("%s=%s", rockName, rockVersion)
+				versionFileLines = append(versionFileLines, rockLine)
+			}
+		}
+	}
+
+	versionFilePath := filepath.Join(appDirPath, versionFileName)
+	versionFile, err := os.Create(versionFilePath)
+	if err != nil {
+		return fmt.Errorf("Failed to write VERSION file %s: %s", versionFilePath, err)
+	}
+
+	defer versionFile.Close()
+
+	versionFile.WriteString(strings.Join(versionFileLines, "\n") + "\n")
+
+	return nil
+}
+
+func getRocksVersions(appDirPath string, projectCtx *project.ProjectCtx) (rocksVersionsMapType, error) {
+	rocksVersionsMap := rocksVersionsMapType{}
+
+	manifestFilePath := filepath.Join(appDirPath, ".rocks/share/tarantool/rocks/manifest")
+	if _, err := os.Stat(manifestFilePath); err == nil {
+		L := lua.NewState()
+		defer L.Close()
+
+		if err := L.DoFile(manifestFilePath); err != nil {
+			return nil, fmt.Errorf("Failed to read manifest file %s: %s", manifestFilePath, err)
+		}
+
+		depsL := L.Env.RawGetString("dependencies")
+		depsLTable, ok := depsL.(*lua.LTable)
+		if !ok {
+			return nil, fmt.Errorf("Failed to read manifest file: dependencies is not a table")
+		}
+
+		depsLTable.ForEach(func(depNameL lua.LValue, depInfoL lua.LValue) {
+			depName := depNameL.String()
+
+			depInfoLTable, ok := depInfoL.(*lua.LTable)
+			if !ok {
+				log.Warnf("Failed to get %s dependency info", depName)
+			} else {
+				depInfoLTable.ForEach(func(depVersionL lua.LValue, _ lua.LValue) {
+					depVersion := depVersionL.String()
+					if _, found := rocksVersionsMap[depName]; found {
+						log.Warnf(
+							"Found multiple versions for %s dependency in rocks manifest",
+							depName,
+						)
+					}
+					rocksVersionsMap[depName] = depVersion
+				})
+			}
+		})
+
+	} else if !os.IsNotExist(err) {
```

Comment:
And handle other errors in else section

Reply:
Again...

Reply:
But stop, I handle all other errors here and ignore manifest file absence.

Reply:
Oh, yes, my bad.